src/ynETH.sol

```diff
@@ -217,7 +265,13 @@ contract ynETH is IynETH, ynBase, IYnETHEvents {
     /// @notice Processes ETH that has been withdrawn from the staking nodes and adds it to the pool.
     /// @dev This function can only be called by the Staking Nodes Manager.
     /// It increases the total deposited in the pool by the amount of ETH sent with the call.
-    function processWithdrawnETH() public payable onlyStakingNodesManager {
+    function processWithdrawnETH() public payable {
+
+        // ETH can be returned either by the stakinNodesManager or by the redemptionAssetsVault
+        if (!(msg.sender == address(stakingNodesManager)
+            || msg.sender == (address(stakingNodesManager.redemptionAssetsVault())))) {
+
```

Comment:
there is nothing happening here, you are missing a revert in the if-body. Right now anybody can call `processWithdrawnETH` (not that that is such a big problem, but it shouldn't be allowed)

Also, just a personal preference but i think the if-condition is more readable if you write it like:

```solidity
if (msg.sender != address(stakingNodesManager &&
    msg.sender != address(stakingNodesManager.redemptionAssetsVault())) {
```